### PR TITLE
feat(kiro): add native SDD subagents and dedicated Kiro model config flow

### LINKED ISSUE #166: feat(kiro): agregar soporte para agente/ide KIRO
### Pre-flight Checklist

- [x] I have searched [existing issues](https://github.com/Gentleman-Programming/gentle-ai/issues) and this is not a duplicate
- [x] I understand that PRs will be rejected if the linked issue does not have `status:approved`

### 🔍 Affected Area

CLI (commands, flags)

### 💡 Problem Statement

Siguiendo la línea de las peticiones anteriores, quería solicitar el soporte para el IDE Kiro de AWS.
Creo que, de los que aún no están soportados, este es importante para un nicho especifico. Actualmente, muchas empresas, por políticas de seguridad y al ser partners directos de AWS, tienen restringida la descarga de otros IDEs del mercado. Por temas contractuales y de cumplimiento, el único entorno que se nos permite usar es Kiro debido a su soporte nativo y oficial con AWS.
Contar con esta integración sería una mejora crítica para quienes trabajamos en estos entornos corporativos restringidos y controlados, pero que dejan tener compatibles con proyectos opensources como estos.

### 📦 Proposed Solution

La idea es instalarlo de alguna manera para que funcione en el IDE o por CLI de KIRO, debería ser parecido a VSCode supongo o algún otro editor en los que ya han tenido casos parecidos.

### 🔄 Alternatives Considered

_No response_

### 📎 Additional Context

_No response_

### PR DESCRIPTION
**Pull request by SantiagoGaonaC**
## 🔗 Linked Issue
Closes #166
---
## 🏷️ PR Type
What kind of change does this PR introduce?
- [x] `type:feature` — New feature (non-breaking change that adds functionality)
---
## 📝 Summary
This PR adds first-class Kiro IDE support aligned with the current gentle-ai SDD architecture:
- Native SDD subagents in `~/.kiro/agents/` for all phases
- Steering strategy integration in `~/.kiro/steering/gentle-ai.md`
- Dedicated Kiro model configuration flow in TUI (`Configure Kiro models`) with independent per-phase assignments
- End-to-end sync/inject plumbing so selected Kiro models update `model:` in generated Kiro subagent frontmatter
- Windows compatibility fixes for skills path resolution and cross-platform path guidance in Kiro SDD subagent prompts
Credit: this extends the initial Kiro groundwork by @calvimor.
---
## 📂 Changes
| File / Area | What Changed |
|-------------|-------------|
| `internal/agents/kiro/adapter.go` | Added native subagent support methods; fixed `SkillsDir()` to `~/.kiro/skills/` |
| `internal/model/kiro_model.go` | Added/updated alias→Kiro model ID mapping (`opus/sonnet/haiku`) |
| `internal/components/sdd/inject.go` | Added Kiro model placeholder resolution and Kiro-specific assignment support |
| `internal/assets/kiro/agents/*.md` | Added/updated 10 Kiro SDD subagent templates |
| `internal/assets/kiro/sdd-orchestrator.md` | Updated to delegation model language |
| `internal/tui/screens/model_config.go` | Added `Configure Kiro models` option (ordered after OpenCode) |
| `internal/tui/screens/kiro_model_picker.go` | New dedicated Kiro model assignment screen/state |
| `internal/tui/model.go` | Added Kiro picker flow, navigation, and sync override wiring |
| `internal/model/selection.go` | Added `KiroModelAssignments` to Selection/SyncOverrides |
| `internal/cli/run.go`, `internal/cli/sync.go`, `internal/app/app.go` | Plumbed Kiro assignments through run/sync/app override pipeline |
| `internal/assets/skills/_shared/SKILL.md` | Added metadata placeholder for Kiro skill scanner compatibility |
| `docs/kiro.md`, `docs/agents.md`, `README.md` | Updated docs to reflect native Kiro subagents and behavior |
| `testdata/golden/sdd-kiro-*.golden` | Regenerated Kiro SDD goldens |
| `*_test.go` (model/agents/components/tui/app) | Added/updated tests for adapter, injection, TUI flow, and path/model behavior |
---
## 🧪 Test Plan
**Unit Tests**
```bash
go test ./...
```
**E2E Tests** (Docker required)
```bash
cd e2e && ./docker-test.sh
```
- [x] Unit tests pass (`go test ./...`)
- [x] E2E tests pass (`cd e2e && ./docker-test.sh`)
- [x] Manually tested locally
Additional manual checks:
- Verified Kiro generation under `~/.kiro/agents/`, `~/.kiro/steering/`, and `~/.kiro/skills/`
- Verified `Configure Models -> Configure Kiro models` updates `model:` in generated Kiro subagent files
- Verified Windows MCP file-read path behavior with `%USERPROFILE%\\.kiro\\skills\\...`
---
## ✅ Contributor Checklist
- [x] PR is linked to an issue with `status:approved`
- [x] I have added the appropriate `type:*` label to this PR
- [x] Unit tests pass (`go test ./...`)
- [x] E2E tests pass (`cd e2e && ./docker-test.sh`)
- [x] I have updated documentation if necessary
- [x] My commits follow [Conventional Commits](https://www.conventionalcommits.org/) format
- [x] My commits do not include `Co-Authored-By` trailers
---
## 💬 Notes for Reviewers
- Kiro assignment flow is intentionally separate from Claude/OpenCode in TUI and persisted independently (`KiroModelAssignments`).
- Kiro injection prioritizes `KiroModelAssignments` with backward-compatible fallback to legacy Claude assignments when unset.
- Kiro subagent prompts now include explicit Windows/macOS/Linux skill paths to avoid MCP access errors on Windows.

### CODE REVIEW

File: internal/tui/screens/model_config.go [hunk 1]
```diff
@@ -11,12 +11,13 @@ func ModelConfigOptions() []string {
...
 		"Configure OpenCode models",
+		"Configure Kiro models",
 		"Back",
 	}
 }
 
 // RenderModelConfig renders the model configuration entry screen.
-// It shows a 3-option menu: Claude models, OpenCode models, Back.
+// It shows a 4-option menu: Claude models, Kiro models, OpenCode models, Back.
 // cursor indicates which option is currently highlighted.
```
[the remark this answers is not in the document]
Comment (SantiagoGaonaC) [reply internal/tui/screens/model_config.go:21]: Fixed. Updated the doc comment to match the actual rendered order:
Claude models → OpenCode models → Kiro models → Back.
Commit: 4fec4a8

File: docs/kiro.md [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+
+gentle-ai supports Kiro as a **native-subagent** platform (`kiro-ide`).
+
+When configured, gentle-ai installs:
+
+| Artifact | Path |
+|----------|------|
+| Steering file | `~/.kiro/steering/gentle-ai.md` |
+| Native SDD agents | `~/.kiro/agents/sdd-{phase}.md` *(10 files)* |
+| Skills directory | `<GlobalConfigDir>/skills/` |
```
[the remark this answers is not in the document]
Comment (SantiagoGaonaC) [reply docs/kiro.md]: Fixed. The overview table already showed `~/.kiro/skills/` (corrected in a prior commit).
Verified: the Kiro adapter writes skills to `filepath.Join(homeDir, ".kiro", "skills")`
on all platforms, independent of GlobalConfigDir.

File: docs/platforms.md [hunk 3]
```diff
@@ -59,3 +59,4 @@ If the hash matches `checksums.txt`, the file is authentic for that release.
 | Codex | `%USERPROFILE%\.codex\` |
 | Windsurf | `%USERPROFILE%\.codeium\windsurf\` (skills, MCP, rules) + `%APPDATA%\Windsurf\User\` (settings) |
 | Antigravity | `%USERPROFILE%\.gemini\antigravity\` |
+| Kiro IDE | `%APPDATA%\kiro\User\` (prompts, skills, settings) + `%USERPROFILE%\.kiro\settings\mcp.json` (MCP) |
```
[the remark this answers is not in the document]
Comment (SantiagoGaonaC) [reply docs/platforms.md]: Fixed. Updated the Windows Kiro row to accurately reflect the split-root layout:
- Prompts (steering): %USERPROFILE%\.kiro\steering\
- Skills: %USERPROFILE%\.kiro\skills\
- SDD agents: %USERPROFILE%\.kiro\agents\
- Settings: %APPDATA%\kiro\User\settings.json
- MCP: %USERPROFILE%\.kiro\settings\mcp.json
Commit: 4fec4a8

File: docs/agents.md [hunk 4]
```diff
@@ -111,3 +120,13 @@ Antigravity is an agent-first platform with built-in sub-agents (Browser, Termin
 - System prompt appended to `~/.gemini/GEMINI.md` (shared with Gemini CLI — collision check warns if both are installed)
 - Mission Control handles built-in sub-agent delegation (Browser, Terminal) automatically
 - Settings managed via the IDE's Agent settings UI, not via `settings.json`
+
+### Kiro IDE
+- **Detection**: gentle-ai detects Kiro by resolving `kiro` from `PATH` — the binary must be available
+- **Steering file** (all platforms): `~/.kiro/steering/gentle-ai.md` with frontmatter `inclusion: always`
+- Native subagents at `~/.kiro/agents/sdd-{phase}.md` (10 files)
+- Skills at the platform-specific Kiro User dir under `skills/`
```
[the remark this answers is not in the document]
Comment (SantiagoGaonaC) [reply docs/agents.md]: Already correct — the bullet reads "Skills (all platforms) at `~/.kiro/skills/`".
This was fixed in a prior commit (fix(kiro): make SkillsDir platform-agnostic).

File: internal/assets/skills/_shared/SKILL.md [hunk 5]
```diff
@@ -0,0 +1,18 @@
+---
+name: _shared
+description: >
+  Internal shared references for SDD skills. Not an invokable skill.
+license: MIT
+metadata:
+  author: gentleman-programming
+  version: "1.0"
+---
```
[the remark this answers is not in the document]
Comment (SantiagoGaonaC) [reply internal/assets/skills/_shared/SKILL.md:9]: Fixed. Added "SKILL.md" to the sharedFiles list in inject.go so it is written to
`<skillsDir>/_shared/SKILL.md` alongside the other shared reference files.
Updated integration test counters accordingly.
Commit: 4fec4a8

File: internal/tui/screens/kiro_model_picker.go [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+	state.InCustomMode = bridge.InCustomMode
+	return handled, assignments
+}
+
+func KiroModelPickerOptionCount(state KiroModelPickerState) int {
+	if state.InCustomMode {
+		return len(claudePhases) + 2 // phase rows + Confirm + Back
+	}
+	return len(claudePresetOrder) + 1 // presets + Back
+}
```
[the remark this answers is not in the document]
Comment (SantiagoGaonaC) [reply internal/tui/screens/kiro_model_picker.go:50]: Acknowledged. claudePhases intentionally mirrors the Claude picker's phase list
(which also excludes sdd-init and sdd-onboard) for consistency.
Users can assign a "default" fallback that covers all unlisted phases.
Updated the UI copy to clarify scope: "each SDD execution phase (explore → apply → archive + orchestrator)".
Commit: 4fec4a8

File: internal/components/sdd/inject.go [hunk 7]
```diff
@@ -502,9 +511,32 @@ func Inject(homeDir string, adapter agents.Adapter, sddMode model.SDDModeID, opt
...
+				} else if opts.ClaudeModelAssignments != nil {
+					// Backward-compatible fallback when Kiro-specific assignments are not provided.
+					if a, hasAlias := opts.ClaudeModelAssignments[phase]; hasAlias {
+						alias = a
+					} else if d, hasDefault := opts.ClaudeModelAssignments["default"]; hasDefault {
+						alias = d
+					}
+				}
+				contentStr = strings.ReplaceAll(contentStr, "{{KIRO_MODEL}}", kmr.KiroModelID(alias))
+			}
```
[the remark this answers is not in the document]
Comment (SantiagoGaonaC) [reply internal/components/sdd/inject.go:537]: Fixed. Added TestInjectKiroModelAssignmentsTakePrecedenceOverClaude which passes both
maps with conflicting values for sdd-design and asserts that KiroModelAssignments wins.
Commit: 4fec4a8

File: internal/tui/screens/model_config_test.go [hunk 8]
```diff
@@ -8,20 +8,20 @@ import (
...
-		t.Fatalf("ModelConfigOptions() len = %d, want 3; got %v", len(opts), opts)
+	if len(opts) != 4 {
+		t.Fatalf("ModelConfigOptions() len = %d, want 4; got %v", len(opts), opts)
 	}
 }
 
 // TestModelConfigOptions_ContainsRequiredItems verifies the three options are
-// Claude, OpenCode, and Back (in order).
+// Claude, Kiro, OpenCode, and Back (in order).
 func TestModelConfigOptions_ContainsRequiredItems(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (SantiagoGaonaC) [reply internal/tui/screens/model_config_test.go]: Fixed. Updated both test comments to reflect the actual menu order:
Claude → OpenCode → Kiro → Back.
Commit: 9af9c14

File: internal/tui/screens/model_config_test.go [hunk 9]
```diff
@@ -8,20 +8,20 @@ import (
 // ─── ModelConfigOptions ────────────────────────────────────────────────────
 
 // TestModelConfigOptions_Count verifies that ModelConfigOptions returns exactly
-// 3 items: Claude, OpenCode, and Back.
+// 4 items: Claude, Kiro, OpenCode, and Back.
 func TestModelConfigOptions_Count(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (SantiagoGaonaC) [reply internal/tui/screens/model_config_test.go:12]: Fixed. Updated both test comments to reflect the actual menu order:
Claude → OpenCode → Kiro → Back.
Commit: 9af9c14

File: internal/agents/kiro/adapter.go [hunk 10]
```diff
@@ -0,0 +1,181 @@
...
+//   - Windows: %APPDATA%/kiro/User/
+// System prompts are loaded from .instructions.md files in the prompts folder
+
+func (a *Adapter) GlobalConfigDir(homeDir string) string {
+	return a.kiroConfigDir(homeDir)
+}
+
+func (a *Adapter) SystemPromptDir(homeDir string) string {
+	return filepath.Join(homeDir, ".kiro", "steering")
+}
```
[the remark this answers is not in the document]
Comment (SantiagoGaonaC) [reply internal/agents/kiro/adapter.go:93]: GlobalConfigDir points (%APPDATA%\kiro\User) and where the managed artifacts actually live (~/.kiro). After investigation, backup/rollback already uses the correct paths via componentPaths() (SkillsDir, SystemPromptFile, MCPConfigPath, etc.) — GlobalConfigDir is not involved there.
The actual bug was in config_scan.go, which used the OS app-config dir for Kiro detection. On Windows, %APPDATA%\kiro\User may not exist even when Kiro is installed. Fixed by switching to ~/.kiro as the detection
root (same as all other home-based agents in the scan list).
Commit: 056dd6a

File: internal/agents/kiro/adapter.go [hunk 11]
```diff
@@ -0,0 +1,181 @@
...
+func (a *Adapter) Detect(_ context.Context, _ string) (bool, string, string, bool, error) {
+	// Kiro IDE is a VS Code fork available as a desktop application.
+	// Official website: https://kiro.dev/
+	// Detect by the "kiro" binary on PATH.
+	binaryPath, err := a.lookPath("kiro")
+	if err != nil {
+		return false, "", "", false, nil
+	}
+
+	return true, binaryPath, "", true, nil
```
[the remark this answers is not in the document]
Comment (SantiagoGaonaC) [reply internal/agents/kiro/adapter.go]: Fixed. Aligned Detect() with the pattern used by Cursor and other adapters:
- configPath now returns ~/.kiro (the home-based root where all managed
  artifacts live: steering, skills, agents, MCP)
- configFound now reflects actual filesystem presence of that directory
- Added statPath as injectable dependency (same pattern as Cursor adapter)

> Note: GlobalConfigDir() was NOT used as configPath because it points to
%APPDATA%\kiro\User on Windows — the OS app-config dir used only for
settings.json, not the IDE-readable ~/.kiro root. This is consistent with
the split-root fixes applied earlier in this PR.

Commit: accbee3

File: internal/agents/kiro/adapter.go [hunk 12]
```diff
@@ -0,0 +1,181 @@
...
+func (a *Adapter) InstallCommand(_ system.PlatformProfile) ([][]string, error) {
+	return nil, AgentNotInstallableError{Agent: model.AgentKiroIDE}
+}
+
+// --- Config paths ---
+// Kiro IDE (VS Code fork) uses similar directory structure to VS Code:
+//   - macOS: ~/Library/Application Support/Kiro/User/
+//   - Linux: ~/.config/kiro/user/ (respects XDG_CONFIG_HOME)
+//   - Windows: %APPDATA%/kiro/User/
+// System prompts are loaded from .instructions.md files in the prompts folder
```
[the remark this answers is not in the document]
Comment (SantiagoGaonaC) [reply internal/agents/kiro/adapter.go]: Fixed. Removed the stale reference to ".instructions.md files" and updated the comment to document the actual split-root layout:
- `~/.kiro/` → steering, skills, agents, MCP (all platforms)
- OS app-config dir → `settings.json` only
Also clarifies that steering content is written via StrategySteeringFile to `~/.kiro/steering/gentle-ai.md`.
Commit: 05981cc

File: internal/tui/screens/model_config_test.go [hunk 13]
```diff
@@ -8,20 +8,20 @@ import (
...
-// TestModelConfigOptions_ContainsRequiredItems verifies the three options are
-// Claude, OpenCode, and Back (in order).
+// TestModelConfigOptions_ContainsRequiredItems verifies the four options are
+// Claude, OpenCode, Kiro, and Back (in order).
 func TestModelConfigOptions_ContainsRequiredItems(t *testing.T) {
 	opts := ModelConfigOptions()
 
-	want := []string{"Claude", "OpenCode", "Back"}
+	want := []string{"Claude", "Kiro", "OpenCode", "Back"}
 	for _, w := range want {
```
[the remark this answers is not in the document]
Comment (SantiagoGaonaC) [reply internal/tui/screens/model_config_test.go]: Fixed. Replaced the presence-only check with TestModelConfigOptions_Order, which asserts the exact position of each option:
- [0] Claude
- [1] OpenCode
- [2] Kiro
- [3] Back
This gives a real order guarantee, not just presence.
Commit: 7efe13a

File: internal/components/golden_test.go [hunk 14]
```diff
@@ -340,6 +342,57 @@ func TestGoldenSDD_Windsurf(t *testing.T) {
...
+	// Golden-check a representative SDD skill file.
+	skillsDir := adapter.SkillsDir(home)
+	skillInit := readTestFile(t, filepath.Join(skillsDir, "sdd-init", "SKILL.md"))
+	assertGolden(t, "sdd-kiro-skill-sdd-init.golden", skillInit)
+
+	// Verify ALL expected SDD skill files exist.
+	expectedSkills := []string{
+		"sdd-init", "sdd-apply", "sdd-archive", "sdd-explore",
+		"sdd-propose", "sdd-spec", "sdd-design", "sdd-tasks", "sdd-verify",
+	}
```
[the remark this answers is not in the document]
Comment (SantiagoGaonaC) [reply internal/components/golden_test.go]: fix in commit: f6e1993

File: internal/tui/model.go [hunk 15]
```diff
@@ -1166,6 +1223,22 @@ func (m Model) confirmSelection() (tea.Model, tea.Cmd) {
...
+			if m.ModelConfigMode {
+				m.ModelConfigMode = false
+				m.setScreen(ScreenModelConfig)
+				return m, nil
+			}
+			if m.Selection.Preset == model.PresetCustom {
+				m.setScreen(ScreenDependencyTree)
+			} else if m.shouldShowClaudeModelPickerScreen() {
+				m.setScreen(ScreenClaudeModelPicker)
+			} else {
```
[the remark this answers is not in the document]
Comment (SantiagoGaonaC) [reply internal/tui/model.go]: fix in commit: f6e1993

File: internal/system/config_scan.go [hunk 16]
```diff
@@ -33,6 +33,7 @@ func knownAgentConfigDirs(homeDir string) []ConfigState {
 		{Agent: "codex", Path: filepath.Join(homeDir, ".codex")},
 		{Agent: "antigravity", Path: filepath.Join(homeDir, ".gemini", "antigravity")},
 		{Agent: "windsurf", Path: filepath.Join(homeDir, ".codeium", "windsurf")},
+		{Agent: "kiro-ide", Path: filepath.Join(homeDir, ".kiro")},
 	}
```
[the remark this answers is not in the document]
Comment (SantiagoGaonaC) [reply internal/system/config_scan.go:43]: Fixed in commit `f6e1993`

**Review by Alan-TheGentleman — APPROVED** [review]
## Review: feat(kiro): add native SDD subagents and dedicated Kiro model config flow

Solid contribution. The adapter follows every established pattern (split-root paths, detection via lookPath, statPath injection for testability, proper sub-agent support). The test coverage is thorough — golden tests for all 10 phase agents, persona, engram MCP, and skills; unit tests for detection edge cases, model resolution, path independence, and TUI navigation.

A few non-blocking observations:

1. **`stringContains` in `adapter_test.go`** — `strings.Contains` from the stdlib does the same thing. The custom implementation is unnecessary.

2. **`applyOverrides` replaces the map, doesn't merge** — `TestApplyOverrides_KiroModelAssignments` claims "merged independently" but the override replaces the whole `KiroModelAssignments` map, losing any pre-existing phase entries. Same pattern as `ClaudeModelAssignments` so not a regression, but the test should either assert replacement behavior honestly or the code should merge keys.

3. **Cursor "10 agent files" in `docs/agents.md`** — Double-check this is intentional for Cursor (not just Kiro).

4. **Kiro multi-mode** — The dedicated Kiro model picker assigns different models per phase, which is effectively multi-mode SDD. The table in `docs/agents.md` shows Kiro as "No" for multi-mode. Worth a footnote or update.

Everything else looks clean. Tests are meaningful, naming is consistent, the TUI flow integrates naturally alongside the existing Claude/OpenCode pickers, and the documentation is comprehensive. Nice work.

**Comment by Alan-TheGentleman** [comment]
All checks passed, but merging #282 and #265 just introduced conflicts. Please rebase against main:

```bash
git fetch origin main
git rebase origin/main
git push --force-with-lease
```

Once the conflicts are resolved, we'll merge right away — the review is approved and all checks were green.

File: internal/assets/kiro/persona-gentleman.md [hunk 17]
```diff
@@ -0,0 +1,51 @@
+## Rules
+
+- Never add "Co-Authored-By" or AI attribution to commits. Use conventional commits only.
+- Never build after changes.
+- When asking a question, STOP and wait for response. Never continue or assume answers.
+- Never agree with user claims without verification. Say "dejame verificar" and check code/docs first.
+- If user is wrong, explain WHY with evidence. If you were wrong, acknowledge with proof.
+- Always propose alternatives with tradeoffs when relevant.
```
[the remark this answers is not in the document]
Comment (SantiagoGaonaC) [reply internal/assets/kiro/persona-gentleman.md:8]: Fixed. Registered AgentKiroIDE in personaContent() to explicitly readkiro/persona-gentleman.md instead of falling through to the generic fallback. The asset content is identical today — intentionally — but now Kiro has
ts own wired path that can diverge independently without touching Claude, OpenCode, or generic. No dead-weight.
Commit: cba6930

File: internal/components/golden_test.go [hunk 18]
```diff
@@ -340,6 +342,61 @@ func TestGoldenSDD_Windsurf(t *testing.T) {
...
+			t.Errorf("expected SDD skill file %q not found: %v", name, err)
+		}
+	}
+	if _, err := os.Stat(filepath.Join(skillsDir, "_shared", "SKILL.md")); err != nil {
+		t.Errorf("expected SDD shared marker %q not found: %v", filepath.Join("_shared", "SKILL.md"), err)
+	}
+
+	// Verify all 10 Kiro native SDD phase agent files with golden snapshots.
+	// SubAgentsDir is not part of the agents.Adapter interface — use the concrete adapter.
+	agentsDir := filepath.Join(home, ".kiro", "agents")
```
[the remark this answers is not in the document]
Comment (SantiagoGaonaC) [reply internal/components/golden_test.go]: Fixed. Replaced the hard-coded filepath.Join(home, ".kiro", "agents") with a type-assertion to a local subAgentDirProvider interface and SubAgentsDir(home). The test now fails fast with a clear message if the adapter ever stopsimplementing SubAgentsDir, and stays correct if the path changes.
Commit: cba6930

File: internal/tui/model.go [hunk 19]
```diff
@@ -1919,6 +2004,15 @@ func (m Model) goBack() Model {
...
 	}
 
+	if m.Screen == ScreenKiroModelPicker && m.Selection.Preset == model.PresetCustom {
+		if m.shouldShowClaudeModelPickerScreen() {
+			m.setScreen(ScreenClaudeModelPicker)
+		} else {
+			m.setScreen(ScreenDependencyTree)
+		}
+		return m
+	}
```
[the remark this answers is not in the document]
Comment (SantiagoGaonaC) [reply internal/tui/model.go]: Fixed. Extended goBack() to handle the non-custom case for ScreenKiroModelPicker with the same logic as Enter on "← Back":
- Claude in flow → ScreenClaudeModelPicker
- No Claude → ScreenPreset
Added two regression tests (TestKiroPickerEscNonCustomWithClaudeGoesToClaudePicker and TestKiroPickerEscNonCustomWithoutClaudeGoesToPreset) to lock this behavior.
Commit: 4bb3cb0

File: internal/agents/kiro/adapter.go [hunk 20]
```diff
@@ -0,0 +1,198 @@
...
+	// Note: configPath is ~/.kiro (the home-based root where all managed
+	// artifacts live), NOT GlobalConfigDir() which points to the OS app-config
+	// dir (%APPDATA%\kiro\User on Windows) used only for settings.json.
+	configPath := filepath.Join(homeDir, ".kiro")
+
+	binaryPath, err := a.lookPath("kiro")
+	if err != nil {
+		// Binary not found — Kiro is not installed.
+		return false, "", configPath, false, nil
+	}
```
[the remark this answers is not in the document]
Comment (SantiagoGaonaC) [reply internal/agents/kiro/adapter.go:60]: Fixed. Detect() now uses errors.Is(err, exec.ErrNotFound) to distinguish "not installed" from unexpected failures. IO/permission errors are surfaced as non-nil. Added TestAdapter_Detect_UnexpectedLookPathError.
Commit: 8b08786

File: internal/components/sdd/inject_test.go [hunk 21]
```diff
@@ -2937,6 +2937,147 @@ func TestInjectCursorWritesSubAgentFiles(t *testing.T) {
...
+		// Non-default overrides we need to prove at runtime.
+		"sdd-design":  model.ClaudeModelOpus,
+		"sdd-archive": model.ClaudeModelHaiku,
+		// Default fallback for unspecified phases.
+		"default": model.ClaudeModelSonnet,
+	}
+
+	result, err := Inject(home, adapter, "", InjectOptions{ClaudeModelAssignments: assignments})
+	if err != nil {
+		t.Fatalf("Inject(kiro, custom assignments) error = %v", err)
```
[the remark this answers is not in the document]
Comment (SantiagoGaonaC) [reply internal/components/sdd/inject_test.go]: Fixed. Renamed to TestInjectKiroFallsBackToClaudeModelAssignmentsWhenKiroMapUnset to clearly document that this tests the legacy ClaudeModelAssignments fallback, not the KiroModelAssignments path. Commit: 8b08786

File: docs/kiro.md [hunk 22]
```diff
@@ -0,0 +1,159 @@
...
+## Detection
+
+gentle-ai detects Kiro by resolving `kiro` from `PATH`:
+
+```
+exec.LookPath("kiro")
+```
+
+If `kiro` is not on `PATH`, Kiro will **not** be auto-detected during install. Make sure the Kiro binary is accessible in your shell environment before running `gentle-ai install`.
+
```
[the remark this answers is not in the document]
Comment (SantiagoGaonaC) [reply docs/kiro.md:35]: Fixed. Updated both docs to reflect the actual detection flow: install/TUI uses ~/.kiro presence (system.ScanConfigs), PATH-based detection applies only to sync/upgrade via adapter.Detect().
Commit: 8b08786

File: docs/agents.md [hunk 23]
```diff
@@ -111,3 +122,13 @@ Antigravity is an agent-first platform with built-in sub-agents (Browser, Termin
 - System prompt appended to `~/.gemini/GEMINI.md` (shared with Gemini CLI — collision check warns if both are installed)
 - Mission Control handles built-in sub-agent delegation (Browser, Terminal) automatically
 - Settings managed via the IDE's Agent settings UI, not via `settings.json`
+
+### Kiro IDE
+- **Detection**: gentle-ai detects Kiro by resolving `kiro` from `PATH` — the binary must be available
```
[the remark this answers is not in the document]
Comment (SantiagoGaonaC) [reply docs/agents.md]: Fixed. Detect() now uses errors.Is(err, exec.ErrNotFound) to distinguish "not installed" from unexpected failures. IO/permission errors are surfaced as non-nil. Added TestAdapter_Detect_UnexpectedLookPathError.
Commit: 8b08786

File: docs/kiro.md [hunk 24]
```diff
@@ -0,0 +1,159 @@
...
+---
+
+## ⚠️ MCP Path Separation
+
+Kiro stores user config and MCP config in **different root directories**.
+
+- **User config** (prompts, skills, settings) lives under the platform-native Kiro User dir (`~/Library/Application Support/Kiro/User`, `%APPDATA%\kiro\User`, or `$XDG_CONFIG_HOME/kiro/user`)
+- **MCP config** is always at `~/.kiro/settings/mcp.json` (or `%USERPROFILE%\.kiro\settings\mcp.json` on Windows), regardless of platform
+
+If MCP tools are not loading, check the **separate `.kiro/settings/` root** rather than the Kiro User config dir.
```
[the remark this answers is not in the document]
Comment (SantiagoGaonaC) [reply docs/kiro.md]: Fixed. Renamed section to "Split-Root Layout" and corrected the description. Steering/skills/agents live under ~/.kiro/ on all platforms. The platform-native User dir is only for settings.json. Commit: 8b08786

File: internal/app/app.go [hunk 25]
```diff
@@ -310,6 +310,9 @@ func applyOverrides(selection *model.Selection, overrides *model.SyncOverrides)
 	if overrides.ClaudeModelAssignments != nil {
 		selection.ClaudeModelAssignments = overrides.ClaudeModelAssignments
 	}
+	if overrides.KiroModelAssignments != nil {
+		selection.KiroModelAssignments = overrides.KiroModelAssignments
+	}
```
[the remark this answers is not in the document]
Comment (SantiagoGaonaC) [reply internal/app/app.go:319]: Fixed. Added kiro_model_assignments to state.InstallState and wired it through persistAssignments() + loadPersistedAssignments() using the same pattern as ClaudeModelAssignments. User Kiro model selections now survive across sync runs. Added round-trip test. Commit: 9973cf9

File: internal/cli/sync.go [hunk 26]
```diff
@@ -492,6 +492,7 @@ func (s componentSyncStep) Run() error {
 			opts := sdd.InjectOptions{
 				OpenCodeModelAssignments: s.selection.ModelAssignments,
 				ClaudeModelAssignments:   s.selection.ClaudeModelAssignments,
+				KiroModelAssignments:     s.selection.KiroModelAssignments,
 				WorkspaceDir:             s.workspaceDir,
 				StrictTDD:                s.selection.StrictTDD,
 				Profiles:                 profiles,
```
[the remark this answers is not in the document]
Comment (SantiagoGaonaC) [reply internal/cli/sync.go:498]: Fixed. Added kiro_model_assignments to state.InstallState and wired it through persistAssignments() + loadPersistedAssignments() using the same pattern as ClaudeModelAssignments. User Kiro model selections now survive across sync runs. Added round-trip test. Commit: 9973cf9

File: internal/components/sdd/inject.go [hunk 27]
```diff
@@ -885,6 +920,10 @@ func injectFileAppend(homeDir string, adapter agents.Adapter) (InjectionResult,
 		existing = instructionsFrontmatter
 	}
 
+	if adapter.SystemPromptStrategy() == model.StrategySteeringFile && strings.TrimSpace(existing) == "" {
+		existing = steeringFrontmatter
+	}
+
 	// Use agent-specific SDD orchestrator content when available; fall back to generic.
 	content := assets.MustRead(sddOrchestratorAsset(adapter.Agent()))
```
[the remark this answers is not in the document]
Comment (SantiagoGaonaC) [reply internal/components/sdd/inject.go:931]: Deferred by design. The steering file (gentle-ai.md) is the orchestrator persona — it should not double as a model config summary. The selected models are already stamped directly into each ~/.kiro/agents/sdd-{phase}.md frontmatter at sync time, which is the canonical source of truth for per-phase model assignments in Kiro.

**Review by Alan-TheGentleman — APPROVED** [review]
## Re-Review: All Suggestions Addressed

Re-reviewed the 24 new commits. Checking the 4 items from the previous review:

| # | Suggestion | Status |
|---|-----------|--------|
| 1 | Replace custom `stringContains` with `strings.Contains` | Fixed — `649945e` |
| 2 | Fix `applyOverrides` test name/comment (replace != merge) | Fixed — comment now says "replaces the entire map", assertion confirms old keys gone |
| 3 | Cursor "9 agent files" -> verify count is 10 | Fixed — updated to 10 in both locations |
| 4 | Kiro multi-mode should be "Yes" not "No" | Fixed — table shows `Yes*` with footnote explaining `KiroModelAssignments` flow |

### Security Audit (66 files)

- No new dependencies (go.mod/go.sum unchanged)
- No binaries, no `go:generate`, no `init()` functions
- No network calls, credentials, or obfuscated code
- `//go:embed` change only adds `all:kiro` (consistent with other agents)

### Notable Additions Since Last Review

- `KiroModelAssignments` persisted in `state.json` (survives across runs)
- `Detect()` properly surfaces unexpected `LookPath` errors
- Esc/Back navigation consistency fixed with regression tests
- `docs/kiro.md` added — thorough and consistent with existing agent docs

LGTM — approving.
